docs/extensibility/60-form-widgets.md

```diff
@@ -0,0 +1,430 @@
+| **defaultExpanded** | | boolean | Specifies if the widget should be expanded by default. Defaults to `false`. |
+
+See the following example:
+
+```yaml
+- path: spec.service
+  widget: FormGroup
+  children:
+    - path: host
+    - path: port
+```
+
+<img src="./assets/form-widgets/FormGroup.png" alt="Example of a FormGroup widget" style="border: 1px solid #D2D5D9">
+
+### `GenericList`
+
+The `GenericList` widgets render an array as a list of collapsible sections with their own sub-forms. An **Add** button is present to add new entries.
+
+These are the available `GenericList` widget parameters:
+
+| Parameter | Required | Type | Description |
+|-----------|----------|------|-------------|
+| **placeholder** | | | Specifies a short hint about the input field value. |
+| **template** | | | Specifies default structure for a list item. |
+| **defaultExpanded** | | boolean | Specifies if the widget should be expanded by default. Defaults to `false`. |
+| **required** | | boolean | Specifies if a field is required. The default value is taken from CRD; if it doesn't exist in the CRD, then it defaults to `false`. |
+
+See the following example:
+
+```yaml
+- path: spec.services
+  widget: GenericList
+  children:
+    - path: '[].host'
+    - path: '[].port'
+
+ - widget: GenericList
+   path: spec.filter.filters
+   children:
+     - path: '[].eventType.value'
+       placeholder: placeholder.eventType
+     - path: '[].eventSource.value'
+   template:
+     eventSource:
+       property: source
+       type: exact
+       value: ''
+     eventType:
+       property: type
+       type: exact
+```
+
+<img src="./assets/form-widgets/GenericList.png" alt="Example of a GenericList widget" style="border: 1px solid #D2D5D9">
+
+### `SimpleList`
+
+The `SimpleList` widgets render an array as a table with rows representing data items and columns representing different fields. New items are added automatically when new entries are typed in.
+
+> **NOTE:** This type of field is only suitable for simple data types and can't contain more complex structures in its items.
+
+These are the available `SimpleList` widget parameters:
+
+| Parameter | Required | Type | Description |
+|-----------|----------|------|-------------|
+| **placeholder** | | | Specifies a short hint about the input field value. |
+| **required** | No | boolean | Specifies if a field is required. The default value is taken from CRD; if it doesn't exist in the CRD, then it defaults to `false`. |
+| **inputInfo** | | string | A string below the input field that shows how to fill in the input. You can use the {{ [`name`] (`link`) }} format to display a `name` instead of a `link. |
+| **description** | | string | A string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD). |
+| **defaultExpanded** | | boolean | Specifies if the widget should be expanded by default. Defaults to `false`. |
```

Comment:
```suggestion
| **placeholder** | No | string | Specifies a short hint about the input field value. |
| **required** | No | boolean | Specifies if a field is required. The default value is taken from CRD; if it doesn't exist in the CRD, then it defaults to `false`. |
| **inputInfo** | No | string | A string below the input field that shows how to fill in the input. You can use the {{ [`name`] (`link`) }} format to display a `name` instead of a `link. |
| **description** | No | string | A string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD). |
| **defaultExpanded** | No | boolean | Specifies if the widget should be expanded by default. Defaults to `false`. |
```